api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

```diff
@@ -125,6 +125,10 @@
     checkNotNull(getter, "getter");
 
     SpanContext spanContext = extractImpl(carrier, getter);
+    if (!spanContext.isValid()) {
```

Comment:
Can we change it so propagator work on `SpanContext`. Is there a reason we use `Context` and copy all this logic into every propagator?

Reply:
I think this arose from the desire to keep context propagation as a separate concern from tracing. I think, looking back on it now, that this might have been a goal that's more academic than useful for end users.

Reply:
> I think this arose from the desire to keep context propagation as a separate concern from tracing.

This is the case.

Reply:
Doesn't have to be this PR but recommend revisit this. The propagators are in a package called `trace` so decoupling from tracing doesn't seem to make much sense, or otherwise the package is wrong.

Reply:
Please open an issue if you think this is important and the reasons. Personally I think there's no problem, but I might be losing some angle ;) (also, see the linked issue).

Reply:
Thanks will discuss this separately, you're right it doesn't need to be taken care of here.